x/stake/keeper/validator.go

```diff
@@ -407,10 +411,8 @@ func (k Keeper) UpdateBondedValidatorsFull(ctx sdk.Context) {
 
 		if validator.Revoked && validator.Status() == sdk.Bonded {
 			panic(fmt.Sprintf("revoked validator cannot be bonded, address: %v\n", ownerAddr))
-		} else {
-			bondedValidatorsCount++
 		}
-
+		bondedValidatorsCount++
```

Comment:
Hmm, what if the validator is revoked - we should make sure it's unbonded, right? (not sure the logic on line 409 is right either)

Seems like we might try to bond a revoked validator and then panic

Reply:
addressed